### DISCUSSION TITLE
2023-04-07 bugfix release

### DISCUSSION THREAD

**Discussion by renatolfc** [microsoft/farmvibes-ai#83] [Announcements]
Yesterday we updated FarmVibes.AI with fixes to the both the backend and the client library. With these fixes, workflow management will make more sense now. We also added a new feature that will improve quality of life when running FarmVibes.AI: the ability to resubmit workflows for execution! Below you will find a summary of all changes.

As usual, if you already have cloned the repository:
 
```
git pull
```
 
should fetch the new code. We recommend an update of the library and images with:
 
```
bash farmvibes-ai.sh update
```
 
Please let us know if you have any issues.

---

## Notebooks

- [⚒️ FIX] We've updated the baseline and scenario files for the Carbon sequestration example notebook, adding a more comprehensive set of examples, including fertilization and organic amendment examples. In addition to that, we have included a more concise set of dates to comply with the current crop year.

## Client

- [⚒️ FIX] We resolved the issue with the monitor showing negative times by correctly adjusting displayed times to the time zone of the client running the monitor.
- [🎉 NEW] We now display warning messages in the monitor, such as when the system is running low on space, or when we can't determine the user's time zone.
- [🎉 NEW] We added support to resubmitting runs with the same input parameters via client (`FarmvibesAiClient().resubmit_run(run_id)`) or directly through the run object (`VibeWorkflowRun().resubmit()`).

## CLI

 - [📈 IMPROVEMENT] The dapr sidecar components are now created by a TerraForm script.

## Backend

 - [⚒️ FIX] We fixed a bug in the worker in which failed tasks would be marked as completed successfully, but without any output. Now these tasks are correctly marked as failed.
 - [🎉 NEW] The backend now supports more general types (`BaseVibe`) that don't require a geometry and a time range to be processed by the system.
 - [⚒️ FIX] We've improved how we download Sentinel data from the Planetary Computer, failing download operations that would otherwise hang in previous versions.
 - [⚒️ FIX] We've fixed a resource leakage in our orchestrator that would make it consume unbounded amounts of memory.
 - [📈 IMPROVEMENT] We've changed our default logging level from `DEBUG` to `INFO`, potentially reducing the amount of disk space used in user deployments.

## Known Issues

- [🐛 BUG] Workflows that are submitted soon after the cluster is started (after a call to `farmvibes-ai.sh stop` followed by a call to `farmvibes-ai.sh start`) might fail, and may require a call to `farmvibes-ai.sh restart` to restore cluster state. This happens because the kubernetes implementation we use reports deployments being in the `Ready` state without that being the case. We are investigating how that can be fixed.

**Comment by richstep** [comment]
running `bash farmvibes-ai.sh update` gives me this error

![image](https://user-images.githubusercontent.com/733563/231321794-b6356c65-3f3f-4e22-a8c4-02f652648de7.png)

Plan: 16 to add, 0 to change, 0 to destroy.
module.kubernetes.kubernetes_namespace.kubernetesdaprnamespace: Creating...
module.kubernetes.kubernetes_persistent_volume.user_storage_pv: Creating...
module.kubernetes.helm_release.redis: Creating...
module.kubernetes.helm_release.rabbitmq: Creating...
╷
│ Error: namespaces "dapr-system" already exists
│
│   with module.kubernetes.kubernetes_namespace.kubernetesdaprnamespace,
│   on modules/kubernetes/dapr.tf line 1, in resource "kubernetes_namespace" "kubernetesdaprnamespace":
│    1: resource "kubernetes_namespace" "kubernetesdaprnamespace" {
│
╵
╷
│ Error: persistentvolumes "user-storage-pv" already exists
│
│   with module.kubernetes.kubernetes_persistent_volume.user_storage_pv,
│   on modules/kubernetes/persistentvolume.tf line 1, in resource "kubernetes_persistent_volume" "user_storage_pv":
│    1: resource "kubernetes_persistent_volume" "user_storage_pv" {
│
╵
╷
│ Error: cannot re-use a name that is still in use
│
│   with module.kubernetes.helm_release.rabbitmq,
│   on modules/kubernetes/rabbitmq.tf line 1, in resource "helm_release" "rabbitmq":
│    1: resource "helm_release" "rabbitmq" {
│
╵
╷
│ Error: cannot re-use a name that is still in use
│
│   with module.kubernetes.helm_release.redis,
│   on modules/kubernetes/redis.tf line 1, in resource "helm_release" "redis":
│    1: resource "helm_release" "redis" {
